docs/filesystem-pv-disks.md

```diff
@@ -0,0 +1,136 @@
+# Using file-system based PersistentVolumes as virtual disk backends
+
+## Introduction
+
+So far Kubernetes PersistentVolumes are remote file-systems attached to a pod at
+runtime. This is currently the only way of exposing remote or local storage to
+pods, through a file-system.
+Despite the fact that storage is exposed as file-systems to pod, KubeVirt has
+no mechanism to use these plain file-system based PVs to act as backends for
+virtual disks.
+
+
+## Goals
+
+This proposal is about adding a simple mechanism to allow using file-system
+based PersistentVolumes for storing file images backing virtual disks.
+
+The focus is primarily on providing the right cluster levels mechanics to enable
+future changes for improving usability, performance, or other characteristics.
+
+
+## Non-Goals
+
+- Provide a totally atumatic solution
+- Cover snapshots explicitly
+
+
+## Use-case
+
+The primary use-case is to allow a VM to use PVs as backend storage for their
+virtual disks.
+
+
+## Design Overview
+
+Currently KubeVirt only supports [direct PV disks](direct-pv-disks.md).
+This feature works, by assigning PVCs backed by network block storage (currently
+only iSCSI) to a VM. The connection details of the PVC are then leveraged with
+qemu's built-in network storage drivers, to directly connect qemu to the remote
+storage. This effectively bypasses all other Kubernetes or KubeVirt components,
+and establishes a direct connection between qemu and the storage backend.
+
+The mechanism proposed in proposal however is leveraging Kubernetes to attach
+the remote storage as a file-system to a pod, to use this file-system to storage
+disk images which are acting as a backend storage to the VM.
+
+
+## API
+
+The general API to use PersistentVolume claims as virtual disk backends was
+introduced with [direct PV proposal](direct-pv-disks.md).
+
+This proposal is merely adding an additional field to support addressing files
+on a file-system.
+
+Thus, to use a PV as a virtual disk backend, a user needs to create a claim for
+the required PV, and then reference the to be used disk image using the newly
+introduced `file` parameter.
+The `file` field takes a path, relative to the source of the file-system held by
+the referenced PVC.
+
+An example:
+
+```yaml
+kind: VM
+spec:
+  domain:
+    devices:
+      disks:
+      - type: PersistentVolumeClaim
+        source:
+          name: vm-01-disks
+          file: disk-01.img  # The change
```

Comment:
I wonder if it is not the right time to introduce something like a storage pool and use relative paths, rather than requiring/allowing absolute paths